### DISCUSSION TITLE
FIM Integration  Who deleted files?

### DISCUSSION THREAD

**Discussion by JAEP2** [Security-Onion-Solutions/securityonion#12483] [2.4]
### Version

2.4.50

### Installation Method

Network installation on Red Hat derivative like Oracle, Rocky, Alma, etc.

### Description

other (please provide detail below)

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

16

### RAM

64

### Storage for /

4TB

### Storage for /nsm

2TB

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Hi all,
Added the FIM integration to test in one Windows system, in other to monitoring unit e:\
Created a new policy cloning previous one:
![imagen](https://github.com/Security-Onion-Solutions/securityonion/assets/31960081/a6dbd909-6bd6-445a-9d7c-3ed7a87c8024)
This policy have the following integrations
![imagen](https://github.com/Security-Onion-Solutions/securityonion/assets/31960081/d72217b5-2116-4877-a688-d1067739fa8f)
I tested several options E:\  E:/ etc
![imagen](https://github.com/Security-Onion-Solutions/securityonion/assets/31960081/aae51e29-e1de-4492-9c35-63a27498a651)
The healthy of the 2 agents with the policy that have FIM integrations shows that are healthy
![imagen](https://github.com/Security-Onion-Solutions/securityonion/assets/31960081/d329889c-f2ba-446f-9c13-549d5dd8a287)
Inputs doesn't show anything, in previous 2.4.40 (we have invalid input) after upgrade healthy ok
![imagen](https://github.com/Security-Onion-Solutions/securityonion/assets/31960081/8f26bcad-2ac4-4c6e-a10b-997657c3e19c)

FIM dashboard, doesn't show any data
![imagen](https://github.com/Security-Onion-Solutions/securityonion/assets/31960081/074f3d03-3085-4c16-a55e-0d869b274ed3)
Thanks a lot in advanced to provided any tip, or instructions to do any test to detect the bug.

Kind regards

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by cm-ops** [comment]
`grep audit /opt/so/saltstack/default/salt/elasticfleet/tools/sbin_jinja/so-elastic-agent-gen-installers` if that does not return anything you may be running into https://github.com/Security-Onion-Solutions/securityonion/pull/12487

**Reply by JAEP2** [reply]
Hi,
The grep returns the following:
**find /nsm/elastic-agent-workspace/elastic-agent-*/data/elastic-agent-*/components -maxdepth 1 -regex '.*fleet.*\|.*packet.*\|.*apm.*\|.*audit.*\|.*heart.*\|.*cloud.*' -delete** 

seems I need to apply a patch..

changing this:

find /nsm/elastic-agent-workspace/elastic-agent-*/data/elastic-agent-*/components -maxdepth 1 -regex '.*fleet.*\|.*packet.*\|.*apm.*\|.*audit.*\|.*heart.*\|.*cloud.*' -delete 

for this
find /nsm/elastic-agent-workspace/elastic-agent-*/data/elastic-agent-*/components -maxdepth 1 -regex '.*fleet.*\|.*packet.*\|.*apm.*\|.*heart.*\|.*cloud.*' -delete 

I will try and check the results

Thanks a lot

**Reply by JAEP2** [reply]
No changes after apply manually  the patch https://github.com/Security-Onion-Solutions/securityonion/pull/12487  still empty dashboards, and everything running ok, and the agents with healthy status... :( any other suggestion? Thanks a lot in advanced...

**Reply by cm-ops** [reply]
You have to regenerate the installers too. Then redeploy the agent.

**Reply by JAEP2** [reply]
Hi cm-ops!,

Works like a charm, thanks a lot

Kind regards

JAEP

**Comment by JAEP2** [comment]
Hi @cm-ops 

Again stop working the FIM policy, empty dashboard again. After apply the patch and regenerate the installer works fine. But after some time stop to send info to dashboardas, health seems ok. I am still using 2.4.50  Could you confirm if FIM works well at 2.4.60?  Thanks in advanced.


NOTE: I  just updated to 2.4.60, FIM works perfect.

**Comment by udi-mosh** [comment]
Hi,
i have encountered the same problem in 2.4.50 but i didn't apply the patch suggested. is change was applied to 2.4.60 ?
i have upgraded to 2.4.60, and reinstalled the customized agent from the installation and FIM does not work.

thanks,
udi

**Reply by udi-mosh** [reply]
what is the correct configuration for fim ?
if i want to monitor windows, is it case sensitive ? do i have to configure c:\ or c:/ ? if i want to configure directory, do i have to end it with / or \ or nothing ?
i have tried everything and i can't get it to work.

**Reply by udi-mosh** [reply]
some update...
i have configured the paths like c:\ and enabled the collect file logs on the FIM integrity config (in 2.4.50 it through an error) and it started working. i also noticed that there were changes pending in the manager's elastic-search component so i initiated a reboot.
upon investigating the logs, i couldn't find who made the change in a file or deleted it. is this something that fim will include ?

**Reply by JAEP2** [reply]
Hi, 
I'm in the same situation, i can`t see who deleted files,  when file is created show the owner, but I didn`t found the way to see who deleted the files... Any suggestion?

Thanks in advanced

**Reply by JAEP2** [reply]
Hi, Upgraded to 2.4.70 rebuild the elastic-agent binarys, and still i did not found the way to see in FIM, wich user deleted the files.

Anyone has been solved this, this is a essential feature for a FIM, we know who created and edited files, but no who deleted.

All the guides to solve it will be apreciated.

Thanks in advanced.

J.A

**Reply by cm-ops** [reply]
I do see this open issue https://github.com/elastic/integrations/issues/3310 in Elastic Integrations GH.
